[PATCH] arm: mediatek: rename nmbm related dts and configs

Renamed several device tree source files and configuration files
associated with NMBM for MT7987 and MT7988 platforms to reflect clarity
and adherence to new naming conventions.
- **Config Files**:
  - Updated defconfig files including `mt7987_spim_nand_nmbm_rfb_defconfig`,
    `mt7988_10g_p0_spim_nand_nmbm_rfb_defconfig`, etc., to ensure consistent
    naming conventions for build configurations.

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
---
 .../dts/{mt7987-spim-nand.dtsi => mt7987-spim-nand-nmbm.dtsi}   | 0
 ...d-rfb-u-boot.dtsi => mt7987a-spim-nand-nmbm-rfb-u-boot.dtsi} | 0
 ...mt7987a-spim-nand-rfb.dts => mt7987a-spim-nand-nmbm-rfb.dts} | 2 +-
 ...0-spim-nand-rfb.dts => mt7988-10g-p0-spim-nand-nmbm-rfb.dts} | 0
 ...1-spim-nand-rfb.dts => mt7988-10g-p1-spim-nand-nmbm-rfb.dts} | 0
 ...spim-nand-rfb.dts => mt7988-i2p5g-p1-spim-nand-nmbm-rfb.dts} | 0
 ...nd-an8831x-rfb.dts => mt7988-spim-nand-nmbm-an8831x-rfb.dts} | 0
 .../{mt7988-spim-nand-rfb.dts => mt7988-spim-nand-nmbm-rfb.dts} | 0
 ...m_nand_rfb_defconfig => mt7987_spim_nand_nmbm_rfb_defconfig} | 2 +-
 ...rfb_defconfig => mt7988_10g_p0_spim_nand_nmbm_rfb_defconfig} | 0
 ...rfb_defconfig => mt7988_10g_p1_spim_nand_nmbm_rfb_defconfig} | 0
 ...b_defconfig => mt7988_i2p5g_p1_spim_nand_nmbm_rfb_defconfig} | 0
 ...fb_defconfig => mt7988_spim_nand_nmbm_an8831x_rfb_defconfig} | 0
 ...m_nand_rfb_defconfig => mt7988_spim_nand_nmbm_rfb_defconfig} | 0
 ..._sb_rfb_defconfig => mt7988_spim_nand_nmbm_sb_rfb_defconfig} | 0
 15 files changed, 2 insertions(+), 2 deletions(-)
 rename arch/arm/dts/{mt7987-spim-nand.dtsi => mt7987-spim-nand-nmbm.dtsi} (100%)
 rename arch/arm/dts/{mt7987a-spim-nand-rfb-u-boot.dtsi => mt7987a-spim-nand-nmbm-rfb-u-boot.dtsi} (100%)
 rename arch/arm/dts/{mt7987a-spim-nand-rfb.dts => mt7987a-spim-nand-nmbm-rfb.dts} (95%)
 rename arch/arm/dts/{mt7988-10g-p0-spim-nand-rfb.dts => mt7988-10g-p0-spim-nand-nmbm-rfb.dts} (100%)
 rename arch/arm/dts/{mt7988-10g-p1-spim-nand-rfb.dts => mt7988-10g-p1-spim-nand-nmbm-rfb.dts} (100%)
 rename arch/arm/dts/{mt7988-i2p5g-p1-spim-nand-rfb.dts => mt7988-i2p5g-p1-spim-nand-nmbm-rfb.dts} (100%)
 rename arch/arm/dts/{mt7988-spim-nand-an8831x-rfb.dts => mt7988-spim-nand-nmbm-an8831x-rfb.dts} (100%)
 rename arch/arm/dts/{mt7988-spim-nand-rfb.dts => mt7988-spim-nand-nmbm-rfb.dts} (100%)
 rename configs/{mt7987_spim_nand_rfb_defconfig => mt7987_spim_nand_nmbm_rfb_defconfig} (98%)
 rename configs/{mt7988_10g_p0_spim_nand_rfb_defconfig => mt7988_10g_p0_spim_nand_nmbm_rfb_defconfig} (100%)
 rename configs/{mt7988_10g_p1_spim_nand_rfb_defconfig => mt7988_10g_p1_spim_nand_nmbm_rfb_defconfig} (100%)
 rename configs/{mt7988_i2p5g_p1_spim_nand_rfb_defconfig => mt7988_i2p5g_p1_spim_nand_nmbm_rfb_defconfig} (100%)
 rename configs/{mt7988_spim_nand_an8831x_rfb_defconfig => mt7988_spim_nand_nmbm_an8831x_rfb_defconfig} (100%)
 rename configs/{mt7988_spim_nand_rfb_defconfig => mt7988_spim_nand_nmbm_rfb_defconfig} (100%)
 rename configs/{mt7988_spim_nand_sb_rfb_defconfig => mt7988_spim_nand_nmbm_sb_rfb_defconfig} (100%)

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7987a-spim-nand-rfb.dts
+++ /dev/null
@@ -1,36 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0 OR MIT)
-
-/dts-v1/;
-#include "mt7987a.dtsi"
-#include "mt7987-spim-nand.dtsi"
-#include <dt-bindings/input/input.h>
-
-/ {
-	model = "MediaTek MT7987A SPIM-NAND RFB";
-	compatible = "mediatek,mt7987a-spim-snand",
-		     "mediatek,mt7987a", "mediatek,mt7987";
-
-	chosen {
-		bootargs = "console=ttyS0,115200n1 loglevel=8  \
-			    earlycon=uart8250,mmio32,0x11000000 \
-			    pci=pcie_bus_perf ubi.block=0,firmware \
-			    root=/dev/fit0 rootwait";
-	};
-	gpio-keys {
-		compatible = "gpio-keys";
-
-		reset {
-			label = "reset";
-			linux,code = <KEY_RESTART>;
-			gpios = <&pio 1 GPIO_ACTIVE_LOW>;
-			debounce-interval = <10>;
-		};
-
-		wps {
-			label = "wps";
-			linux,code = <KEY_WPS_BUTTON>;
-			gpios = <&pio 0 GPIO_ACTIVE_LOW>;
-			debounce-interval = <10>;
-		};
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7987a-spim-nand-nmbm-rfb.dts
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+#include "mt7987a.dtsi"
+#include "mt7987-spim-nand-nmbm.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "MediaTek MT7987A SPIM-NAND RFB";
+	compatible = "mediatek,mt7987a-spim-snand",
+		     "mediatek,mt7987a", "mediatek,mt7987";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n1 loglevel=8  \
+			    earlycon=uart8250,mmio32,0x11000000 \
+			    pci=pcie_bus_perf ubi.block=0,firmware \
+			    root=/dev/fit0 rootwait";
+	};
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <10>;
+		};
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&pio 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <10>;
+		};
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7987_spim_nand_rfb_defconfig
+++ /dev/null
@@ -1,110 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_ENV_OFFSET=0x0
-CONFIG_DEFAULT_DEVICE_TREE="mt7987a-spim-nand-rfb"
-CONFIG_TARGET_MT7987=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_SYS_LOAD_ADDR=0x48000000
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-# CONFIG_MTK_UPGRADE_IMAGE_VERIFY is not set
-CONFIG_ENABLE_NAND_NMBM=y
-# CONFIG_EFI_LOADER is not set
-CONFIG_FIT=y
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7987a-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7987> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_I2C=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x0
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
-CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_IN_MTD=y
-CONFIG_ENV_MTD_DEV="u-boot-env"
-CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-CONFIG_DM_I2C=y
-CONFIG_SYS_I2C_MTK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHY_AIROHA=y
-CONFIG_PHY_AIROHA_AN8801SB=y
-CONFIG_PHY_AIROHA_EN8811H=y
-CONFIG_PHY_MEDIATEK_2P5GE=y
-CONFIG_PHY_ETHERNET_ID=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7987=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_PWM_MTK=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_XZ=y
-CONFIG_HEXDUMP=y

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7987_spim_nand_nmbm_rfb_defconfig
@@ -0,0 +1,110 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x0
+CONFIG_DEFAULT_DEVICE_TREE="mt7987a-spim-nand-nmbm-rfb"
+CONFIG_TARGET_MT7987=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_SYS_LOAD_ADDR=0x48000000
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+# CONFIG_MTK_UPGRADE_IMAGE_VERIFY is not set
+CONFIG_ENABLE_NAND_NMBM=y
+# CONFIG_EFI_LOADER is not set
+CONFIG_FIT=y
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7987a-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7987> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x0
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
+CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_IN_MTD=y
+CONFIG_ENV_MTD_DEV="u-boot-env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MTK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_AIROHA=y
+CONFIG_PHY_AIROHA_AN8801SB=y
+CONFIG_PHY_AIROHA_EN8811H=y
+CONFIG_PHY_MEDIATEK_2P5GE=y
+CONFIG_PHY_ETHERNET_ID=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7987=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_PWM_MTK=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_XZ=y
+CONFIG_HEXDUMP=y

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7987a-spim-nand-nmbm-rfb-u-boot.dtsi
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include "mt7987a-rfb-u-boot.dtsi"
+
+/ {
+	model = "mt7987";
+	compatible = "mediatek,mt7987", "mediatek,mt7987-spim-nand-rfb";
+
+	bl2_verify {
+		bl2_compatible = "spim-nand";
+	};
+};
+
+&spi2 {
+	status = "disabled";
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7987a-spim-nand-rfb-u-boot.dtsi
+++ /dev/null
@@ -1,16 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-
-#include "mt7987a-rfb-u-boot.dtsi"
-
-/ {
-	model = "mt7987";
-	compatible = "mediatek,mt7987", "mediatek,mt7987-spim-nand-rfb";
-
-	bl2_verify {
-		bl2_compatible = "spim-nand";
-	};
-};
-
-&spi2 {
-	status = "disabled";
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7987-spim-nand.dtsi
+++ /dev/null
@@ -1,52 +0,0 @@
-// SPDX-License-Identifier: (GPL-2.0-only OR MIT)
-
-#include "mt7987-pinctrl.dtsi"
-
-&spi0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&spi0_flash_pins>;
-	status = "okay";
-
-	flash@0 {
-		compatible = "spi-nand";
-		reg = <0>;
-		spi-max-frequency = <52000000>;
-		spi-tx-bus-width = <4>;
-		spi-rx-bus-width = <4>;
-		mediatek,nmbm;
-		mediatek,bmt-max-ratio = <1>;
-		mediatek,bmt-max-reserved-blocks = <64>;
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-			partition@0 {
-				label = "BL2";
-				reg = <0x00000 0x0100000>;
-				read-only;
-			};
-			partition@100000 {
-				label = "u-boot-env";
-				reg = <0x0100000 0x0080000>;
-			};
-			factory: partition@180000 {
-				label = "Factory";
-				reg = <0x180000 0x0400000>;
-			};
-			partition@580000 {
-				label = "FIP";
-				reg = <0x580000 0x0200000>;
-			};
-			partition@780000 {
-				label = "ubi";
-				reg = <0x780000 0x7080000>;
-				compatible = "linux,ubi";
-				volumes {
-					ubi_rootfs: ubi-volume-fit {
-						volname = "firmware";
-					};
-				};
-			};
-		};
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7987-spim-nand-nmbm.dtsi
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR MIT)
+
+#include "mt7987-pinctrl.dtsi"
+
+&spi0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_flash_pins>;
+	status = "okay";
+
+	flash@0 {
+		compatible = "spi-nand";
+		reg = <0>;
+		spi-max-frequency = <52000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		mediatek,nmbm;
+		mediatek,bmt-max-ratio = <1>;
+		mediatek,bmt-max-reserved-blocks = <64>;
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			partition@0 {
+				label = "BL2";
+				reg = <0x00000 0x0100000>;
+				read-only;
+			};
+			partition@100000 {
+				label = "u-boot-env";
+				reg = <0x0100000 0x0080000>;
+			};
+			factory: partition@180000 {
+				label = "Factory";
+				reg = <0x180000 0x0400000>;
+			};
+			partition@580000 {
+				label = "FIP";
+				reg = <0x580000 0x0200000>;
+			};
+			partition@780000 {
+				label = "ubi";
+				reg = <0x780000 0x7080000>;
+				compatible = "linux,ubi";
+				volumes {
+					ubi_rootfs: ubi-volume-fit {
+						volname = "firmware";
+					};
+				};
+			};
+		};
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7988-10g-p0-spim-nand-nmbm-rfb.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2022 MediaTek Inc.
+ * Author: Sam Shih <sam.shih@mediatek.com>
+ */
+
+/dts-v1/;
+#include "mt7988-10g-p0-rfb.dts"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "mt7988-rfb";
+	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
+
+	bl2_verify {
+		bl2_compatible = "spim-nand";
+	};
+};
+
+&snand {
+	status = "disabled";
+};
+
+&spi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	must_tx;
+	enhance_timing;
+	dma_ext;
+	ipm_design;
+	support_quad;
+	tick_dly = <2>;
+	sample_sel = <0>;
+
+	spi_nand@0 {
+		compatible = "spi-nand";
+		reg = <0>;
+		spi-max-frequency = <52000000>;
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7988-10g-p0-spim-nand-rfb.dts
+++ /dev/null
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright (c) 2022 MediaTek Inc.
- * Author: Sam Shih <sam.shih@mediatek.com>
- */
-
-/dts-v1/;
-#include "mt7988-10g-p0-rfb.dts"
-#include <dt-bindings/gpio/gpio.h>
-
-/ {
-	model = "mt7988-rfb";
-	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
-
-	bl2_verify {
-		bl2_compatible = "spim-nand";
-	};
-};
-
-&snand {
-	status = "disabled";
-};
-
-&spi0 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	must_tx;
-	enhance_timing;
-	dma_ext;
-	ipm_design;
-	support_quad;
-	tick_dly = <2>;
-	sample_sel = <0>;
-
-	spi_nand@0 {
-		compatible = "spi-nand";
-		reg = <0>;
-		spi-max-frequency = <52000000>;
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7988-10g-p1-spim-nand-nmbm-rfb.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2022 MediaTek Inc.
+ * Author: Sam Shih <sam.shih@mediatek.com>
+ */
+
+/dts-v1/;
+#include "mt7988-10g-p1-rfb.dts"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "mt7988-rfb";
+	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
+
+	bl2_verify {
+		bl2_compatible = "spim-nand";
+	};
+};
+
+&snand {
+	status = "disabled";
+};
+
+&spi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	must_tx;
+	enhance_timing;
+	dma_ext;
+	ipm_design;
+	support_quad;
+	tick_dly = <2>;
+	sample_sel = <0>;
+
+	spi_nand@0 {
+		compatible = "spi-nand";
+		reg = <0>;
+		spi-max-frequency = <52000000>;
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7988-10g-p1-spim-nand-rfb.dts
+++ /dev/null
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright (c) 2022 MediaTek Inc.
- * Author: Sam Shih <sam.shih@mediatek.com>
- */
-
-/dts-v1/;
-#include "mt7988-10g-p1-rfb.dts"
-#include <dt-bindings/gpio/gpio.h>
-
-/ {
-	model = "mt7988-rfb";
-	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
-
-	bl2_verify {
-		bl2_compatible = "spim-nand";
-	};
-};
-
-&snand {
-	status = "disabled";
-};
-
-&spi0 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	must_tx;
-	enhance_timing;
-	dma_ext;
-	ipm_design;
-	support_quad;
-	tick_dly = <2>;
-	sample_sel = <0>;
-
-	spi_nand@0 {
-		compatible = "spi-nand";
-		reg = <0>;
-		spi-max-frequency = <52000000>;
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7988-i2p5g-p1-spim-nand-nmbm-rfb.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2022 MediaTek Inc.
+ * Author: Sam Shih <sam.shih@mediatek.com>
+ */
+
+/dts-v1/;
+#include "mt7988-i2p5g-p1-rfb.dts"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "mt7988-rfb";
+	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
+
+	bl2_verify {
+		bl2_compatible = "spim-nand";
+	};
+};
+
+&snand {
+	status = "disabled";
+};
+
+&spi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	must_tx;
+	enhance_timing;
+	dma_ext;
+	ipm_design;
+	support_quad;
+	tick_dly = <2>;
+	sample_sel = <0>;
+
+	spi_nand@0 {
+		compatible = "spi-nand";
+		reg = <0>;
+		spi-max-frequency = <52000000>;
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7988-i2p5g-p1-spim-nand-rfb.dts
+++ /dev/null
@@ -1,41 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright (c) 2022 MediaTek Inc.
- * Author: Sam Shih <sam.shih@mediatek.com>
- */
-
-/dts-v1/;
-#include "mt7988-i2p5g-p1-rfb.dts"
-#include <dt-bindings/gpio/gpio.h>
-
-/ {
-	model = "mt7988-rfb";
-	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
-
-	bl2_verify {
-		bl2_compatible = "spim-nand";
-	};
-};
-
-&snand {
-	status = "disabled";
-};
-
-&spi0 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	must_tx;
-	enhance_timing;
-	dma_ext;
-	ipm_design;
-	support_quad;
-	tick_dly = <2>;
-	sample_sel = <0>;
-
-	spi_nand@0 {
-		compatible = "spi-nand";
-		reg = <0>;
-		spi-max-frequency = <52000000>;
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/arch/arm/dts/mt7988-spim-nand-nmbm-rfb.dts
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2022 MediaTek Inc.
+ * Author: Sam Shih <sam.shih@mediatek.com>
+ */
+
+/dts-v1/;
+#include "mt7988-rfb.dts"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
+
+	bl2_verify {
+		bl2_compatible = "spim-nand";
+	};
+};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/arch/arm/dts/mt7988-spim-nand-rfb.dts
+++ /dev/null
@@ -1,17 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright (c) 2022 MediaTek Inc.
- * Author: Sam Shih <sam.shih@mediatek.com>
- */
-
-/dts-v1/;
-#include "mt7988-rfb.dts"
-#include <dt-bindings/gpio/gpio.h>
-
-/ {
-	compatible = "mediatek,mt7988-rfb", "mediatek,mt7988-spim-nand-rfb";
-
-	bl2_verify {
-		bl2_compatible = "spim-nand";
-	};
-};

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7988_10g_p0_spim_nand_nmbm_rfb_defconfig
@@ -0,0 +1,104 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x0
+CONFIG_DEFAULT_DEVICE_TREE="mt7988-10g-p0-spim-nand-rfb"
+CONFIG_TARGET_MT7988=y
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_SYS_LOAD_ADDR=0x44000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth2-aqr"
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+CONFIG_ENABLE_NAND_NMBM=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7988-10g-p0-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7988> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
+CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_IN_MTD=y
+CONFIG_ENV_MTD_DEV="u-boot-env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
+CONFIG_PHY_AQUANTIA_FW_ARRAY=y
+CONFIG_PHY_FIXED=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7988=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_PWM_MTK=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_HEXDUMP=y
+# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7988_10g_p0_spim_nand_rfb_defconfig
+++ /dev/null
@@ -1,104 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_ENV_OFFSET=0x0
-CONFIG_DEFAULT_DEVICE_TREE="mt7988-10g-p0-spim-nand-rfb"
-CONFIG_TARGET_MT7988=y
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_SYS_LOAD_ADDR=0x44000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth2-aqr"
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-CONFIG_ENABLE_NAND_NMBM=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7988-10g-p0-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7988> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
-CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_IN_MTD=y
-CONFIG_ENV_MTD_DEV="u-boot-env"
-CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHY_AQUANTIA=y
-CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
-CONFIG_PHY_AQUANTIA_FW_ARRAY=y
-CONFIG_PHY_FIXED=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7988=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_PWM_MTK=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_HEXDUMP=y
-# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7988_10g_p1_spim_nand_nmbm_rfb_defconfig
@@ -0,0 +1,104 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x0
+CONFIG_DEFAULT_DEVICE_TREE="mt7988-10g-p1-spim-nand-rfb"
+CONFIG_TARGET_MT7988=y
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_SYS_LOAD_ADDR=0x44000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth1-aqr"
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+CONFIG_ENABLE_NAND_NMBM=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7988-10g-p1-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7988> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
+CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_IN_MTD=y
+CONFIG_ENV_MTD_DEV="u-boot-env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
+CONFIG_PHY_AQUANTIA_FW_ARRAY=y
+CONFIG_PHY_FIXED=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7988=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_PWM_MTK=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_HEXDUMP=y
+# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7988_10g_p1_spim_nand_rfb_defconfig
+++ /dev/null
@@ -1,104 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_ENV_OFFSET=0x0
-CONFIG_DEFAULT_DEVICE_TREE="mt7988-10g-p1-spim-nand-rfb"
-CONFIG_TARGET_MT7988=y
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_SYS_LOAD_ADDR=0x44000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth1-aqr"
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-CONFIG_ENABLE_NAND_NMBM=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7988-10g-p1-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7988> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
-CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_IN_MTD=y
-CONFIG_ENV_MTD_DEV="u-boot-env"
-CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHY_AQUANTIA=y
-CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
-CONFIG_PHY_AQUANTIA_FW_ARRAY=y
-CONFIG_PHY_FIXED=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7988=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_PWM_MTK=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_HEXDUMP=y
-# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7988_i2p5g_p1_spim_nand_nmbm_rfb_defconfig
@@ -0,0 +1,106 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x0
+CONFIG_DEFAULT_DEVICE_TREE="mt7988-i2p5g-p1-spim-nand-rfb"
+CONFIG_TARGET_MT7988=y
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_SYS_LOAD_ADDR=0x44000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth1-i2p5g-phy"
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+CONFIG_ENABLE_NAND_NMBM=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7988-i2p5g-p1-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7988> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_MII=y
+# CONFIG_CMD_MDIO is not set
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),-(ubi)"
+CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_IN_MTD=y
+CONFIG_ENV_MTD_DEV="u-boot-env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHYLIB_10G=y
+CONFIG_PHY_MEDIATEK_2P5GE=y
+CONFIG_PHY_FIXED=y
+CONFIG_PHY_GIGE=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7988=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_PWM_MTK=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_HEXDUMP=y
+# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7988_i2p5g_p1_spim_nand_rfb_defconfig
+++ /dev/null
@@ -1,106 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_ENV_OFFSET=0x0
-CONFIG_DEFAULT_DEVICE_TREE="mt7988-i2p5g-p1-spim-nand-rfb"
-CONFIG_TARGET_MT7988=y
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_SYS_LOAD_ADDR=0x44000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm#mt7988a-rfb-eth1-i2p5g-phy"
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-CONFIG_ENABLE_NAND_NMBM=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7988-i2p5g-p1-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7988> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_MII=y
-# CONFIG_CMD_MDIO is not set
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),-(ubi)"
-CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_IN_MTD=y
-CONFIG_ENV_MTD_DEV="u-boot-env"
-CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHYLIB_10G=y
-CONFIG_PHY_MEDIATEK_2P5GE=y
-CONFIG_PHY_FIXED=y
-CONFIG_PHY_GIGE=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7988=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_PWM_MTK=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_HEXDUMP=y
-# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7988_spim_nand_nmbm_rfb_defconfig
@@ -0,0 +1,105 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x0
+CONFIG_DEFAULT_DEVICE_TREE="mt7988-spim-nand-rfb"
+CONFIG_TARGET_MT7988=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_SYS_LOAD_ADDR=0x44000000
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm"
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+CONFIG_ENABLE_NAND_NMBM=y
+# CONFIG_EFI_LOADER is not set
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7988-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7988> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
+CONFIG_ENV_OVERWRITE=y
+CONFIG_ENV_IS_IN_MTD=y
+CONFIG_ENV_MTD_DEV="u-boot-env"
+CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
+CONFIG_PHY_AQUANTIA_FW_ARRAY=y
+CONFIG_PHY_MEDIATEK_2P5GE=y
+CONFIG_PHY_FIXED=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7988=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_PWM_MTK=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_HEXDUMP=y

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- /dev/null
+++ b/configs/mt7988_spim_nand_nmbm_sb_rfb_defconfig
@@ -0,0 +1,95 @@
+CONFIG_ARM=y
+CONFIG_SYS_HAS_NONCACHED_MEMORY=y
+CONFIG_POSITION_INDEPENDENT=y
+CONFIG_ARCH_MEDIATEK=y
+CONFIG_TEXT_BASE=0x41e00000
+CONFIG_SYS_MALLOC_F_LEN=0x4000
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_ENV_SIZE=0x20000
+CONFIG_DEFAULT_DEVICE_TREE="mt7988-spim-nand-rfb"
+CONFIG_TARGET_MT7988=y
+CONFIG_DEBUG_UART_BASE=0x11000000
+CONFIG_DEBUG_UART_CLOCK=40000000
+CONFIG_SYS_LOAD_ADDR=0x44000000
+CONFIG_DEBUG_UART=y
+CONFIG_MEDIATEK_BOOTMENU=y
+CONFIG_MEDIATEK_LOAD_FROM_RAM=y
+CONFIG_MTK_WEB_FAILSAFE=y
+CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
+CONFIG_MTK_UBI_SUPPORT=y
+CONFIG_MTK_SECURE_BOOT=y
+CONFIG_MTK_UPGRADE_BL2_VERIFY=y
+CONFIG_MTK_UPGRADE_FIP_VERIFY=y
+CONFIG_ENABLE_NAND_NMBM=y
+CONFIG_SYS_BOOTM_LEN=0x6000000
+CONFIG_AUTOBOOT_MENU_SHOW=y
+CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
+CONFIG_DEFAULT_FDT_FILE="mt7988-spim-nand-rfb"
+CONFIG_SYS_CBSIZE=512
+CONFIG_SYS_PBSIZE=1049
+CONFIG_LOGLEVEL=7
+CONFIG_LOG=y
+CONFIG_SYS_PROMPT="MT7988> "
+# CONFIG_BOOTM_NETBSD is not set
+# CONFIG_BOOTM_PLAN9 is not set
+# CONFIG_BOOTM_RTEMS is not set
+# CONFIG_BOOTM_VXWORKS is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_XIMG is not set
+# CONFIG_CMD_UNLZ4 is not set
+# CONFIG_CMD_UNZIP is not set
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_PWM=y
+CONFIG_CMD_MTD=y
+CONFIG_CMD_NMBM=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_PSTORE=y
+CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
+CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
+CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
+CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
+CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
+CONFIG_CMD_SMC=y
+CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
+CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_USE_IPADDR=y
+CONFIG_IPADDR="192.168.1.1"
+CONFIG_USE_NETMASK=y
+CONFIG_NETMASK="255.255.255.0"
+CONFIG_USE_SERVERIP=y
+CONFIG_SERVERIP="192.168.1.2"
+CONFIG_PROT_TCP=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+# CONFIG_MMC is not set
+CONFIG_DM_MTD=y
+CONFIG_MTD_SPI_NAND=y
+CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_EON=y
+CONFIG_SPI_FLASH_GIGADEVICE=y
+CONFIG_SPI_FLASH_ISSI=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_PHY_FIXED=y
+CONFIG_MEDIATEK_ETH=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_MT7988=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_MTK_POWER_DOMAIN=y
+CONFIG_DM_PWM=y
+CONFIG_RAM=y
+CONFIG_DM_SERIAL=y
+CONFIG_MTK_SERIAL=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_MTK_SPIM=y
+CONFIG_HEXDUMP=y
+# CONFIG_EFI_LOADER is not set

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7988_spim_nand_rfb_defconfig
+++ /dev/null
@@ -1,105 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_ENV_OFFSET=0x0
-CONFIG_DEFAULT_DEVICE_TREE="mt7988-spim-nand-rfb"
-CONFIG_TARGET_MT7988=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_SYS_LOAD_ADDR=0x44000000
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_DEFAULT_FIT_BOOT_CONF="mt7988a-rfb-spim-nand-nmbm"
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-CONFIG_ENABLE_NAND_NMBM=y
-# CONFIG_EFI_LOADER is not set
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7988-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7988> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
-CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_IN_MTD=y
-CONFIG_ENV_MTD_DEV="u-boot-env"
-CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHY_AQUANTIA=y
-CONFIG_PHY_AQUANTIA_UPLOAD_FW=y
-CONFIG_PHY_AQUANTIA_FW_ARRAY=y
-CONFIG_PHY_MEDIATEK_2P5GE=y
-CONFIG_PHY_FIXED=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7988=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_PWM_MTK=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_HEXDUMP=y

Change-Id: I86b0cc2cf91a09925d9bef3da0d990f183a09389
--- a/configs/mt7988_spim_nand_sb_rfb_defconfig
+++ /dev/null
@@ -1,95 +0,0 @@
-CONFIG_ARM=y
-CONFIG_SYS_HAS_NONCACHED_MEMORY=y
-CONFIG_POSITION_INDEPENDENT=y
-CONFIG_ARCH_MEDIATEK=y
-CONFIG_TEXT_BASE=0x41e00000
-CONFIG_SYS_MALLOC_F_LEN=0x4000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_ENV_SIZE=0x20000
-CONFIG_DEFAULT_DEVICE_TREE="mt7988-spim-nand-rfb"
-CONFIG_TARGET_MT7988=y
-CONFIG_DEBUG_UART_BASE=0x11000000
-CONFIG_DEBUG_UART_CLOCK=40000000
-CONFIG_SYS_LOAD_ADDR=0x44000000
-CONFIG_DEBUG_UART=y
-CONFIG_MEDIATEK_BOOTMENU=y
-CONFIG_MEDIATEK_LOAD_FROM_RAM=y
-CONFIG_MTK_WEB_FAILSAFE=y
-CONFIG_MTK_WEB_FAILSAFE_AFTER_BOOT_FAILURE=y
-CONFIG_MTK_UBI_SUPPORT=y
-CONFIG_MTK_SECURE_BOOT=y
-CONFIG_MTK_UPGRADE_BL2_VERIFY=y
-CONFIG_MTK_UPGRADE_FIP_VERIFY=y
-CONFIG_ENABLE_NAND_NMBM=y
-CONFIG_SYS_BOOTM_LEN=0x6000000
-CONFIG_AUTOBOOT_MENU_SHOW=y
-CONFIG_AUTOBOOT_MENU_MTK_SHOW=y
-CONFIG_DEFAULT_FDT_FILE="mt7988-spim-nand-rfb"
-CONFIG_SYS_CBSIZE=512
-CONFIG_SYS_PBSIZE=1049
-CONFIG_LOGLEVEL=7
-CONFIG_LOG=y
-CONFIG_SYS_PROMPT="MT7988> "
-# CONFIG_BOOTM_NETBSD is not set
-# CONFIG_BOOTM_PLAN9 is not set
-# CONFIG_BOOTM_RTEMS is not set
-# CONFIG_BOOTM_VXWORKS is not set
-# CONFIG_CMD_ELF is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_UNLZ4 is not set
-# CONFIG_CMD_UNZIP is not set
-CONFIG_CMD_CLK=y
-CONFIG_CMD_DM=y
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_PWM=y
-CONFIG_CMD_MTD=y
-CONFIG_CMD_NMBM=y
-CONFIG_CMD_SF_TEST=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PSTORE=y
-CONFIG_CMD_PSTORE_MEM_ADDR=0x42ff0000
-CONFIG_CMD_PSTORE_RECORD_SIZE=0x2000
-CONFIG_CMD_PSTORE_CONSOLE_SIZE=0x2000
-CONFIG_CMD_PSTORE_FTRACE_SIZE=0x0
-CONFIG_CMD_PSTORE_PMSG_SIZE=0x2000
-CONFIG_CMD_SMC=y
-CONFIG_MTDIDS_DEFAULT="nmbm0=nmbm0"
-CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),4096k(factory),2048k(fip),115200k(ubi)"
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_USE_IPADDR=y
-CONFIG_IPADDR="192.168.1.1"
-CONFIG_USE_NETMASK=y
-CONFIG_NETMASK="255.255.255.0"
-CONFIG_USE_SERVERIP=y
-CONFIG_SERVERIP="192.168.1.2"
-CONFIG_PROT_TCP=y
-CONFIG_REGMAP=y
-CONFIG_SYSCON=y
-CONFIG_CLK=y
-# CONFIG_MMC is not set
-CONFIG_DM_MTD=y
-CONFIG_MTD_SPI_NAND=y
-CONFIG_DM_SPI_FLASH=y
-CONFIG_SPI_FLASH_EON=y
-CONFIG_SPI_FLASH_GIGADEVICE=y
-CONFIG_SPI_FLASH_ISSI=y
-CONFIG_SPI_FLASH_MACRONIX=y
-CONFIG_SPI_FLASH_SPANSION=y
-CONFIG_SPI_FLASH_STMICRO=y
-CONFIG_SPI_FLASH_WINBOND=y
-CONFIG_PHY_FIXED=y
-CONFIG_MEDIATEK_ETH=y
-CONFIG_PINCTRL=y
-CONFIG_PINCONF=y
-CONFIG_PINCTRL_MT7988=y
-CONFIG_POWER_DOMAIN=y
-CONFIG_MTK_POWER_DOMAIN=y
-CONFIG_DM_PWM=y
-CONFIG_RAM=y
-CONFIG_DM_SERIAL=y
-CONFIG_MTK_SERIAL=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_MTK_SPIM=y
-CONFIG_HEXDUMP=y
-# CONFIG_EFI_LOADER is not set
